client/containers/DashboardSettings/NavBuilder.tsx

```diff
@@ -3,57 +3,64 @@ import { DragDropContext } from 'react-beautiful-dnd';
 import { Button } from '@blueprintjs/core';
 
 import { generateHash } from 'utils/hashes';
+import { CommunityNavigationEntry } from 'client/utils/navigation';
 
-import PageAutocomplete from './PageAutocomplete';
+import PageCollectionAutocomplete from './PageCollectionAutocomplete';
 import NavBuilderList from './NavBuilderList';
 import NavBuilderRow from './NavBuilderRow';
+import { NavBuilderContext } from './navBuilderContext';
 
 require('./navBuilder.scss');
 
-type OwnProps = {
-	initialNav: any[];
-	prefix?: any[];
-	suffix?: any[];
+type Props = {
+	initialNav: CommunityNavgiationEntry[];
+	prefix?: CommunityNavigationEntry[];
+	suffix?: CommunityNavigationEntry[];
 	pages: any[];
+	collections: any[];
 	onChange: (...args: any[]) => any;
 	disableDropdown?: boolean;
 };
 
-const defaultProps = {
-	prefix: [],
-	suffix: [],
-	disableDropdown: false,
-};
-
-type Props = OwnProps & typeof defaultProps;
-
 const NavBuilder = (props: Props) => {
-	const { initialNav, pages, onChange, prefix, suffix, disableDropdown } = props;
-	const [currentNav, setCurrentNav] = useState(initialNav);
+	const {
+		initialNav,
+		pages,
+		collections,
+		onChange,
+		prefix = [],
+		suffix = [],
+		disableDropdown = false,
+	} = props;
+	const [currentNav, setCurrentNav] = useState<CommunityNavigationEntry[]>(initialNav);
 	const userSetElements = currentNav.slice(prefix.length, currentNav.length - suffix.length);
 
-	const reorder = (list, startIndex, endIndex) => {
+	const reorder = (list: CommunityNavigationEntry[], startIndex: number, endIndex: number) => {
 		const result = Array.from(list);
 		const [removed] = result.splice(startIndex, 1);
 		result.splice(endIndex, 0, removed);
-
 		return result;
 	};
+
 	const onDragEnd = (result) => {
 		if (result.destination) {
 			const nextUserElements =
 				result.destination.droppableId === 'main-list'
 					? reorder(userSetElements, result.source.index, result.destination.index)
 					: userSetElements.map((item) => {
-							if (item.id === result.destination.droppableId) {
+							if (
+								typeof item === 'object' &&
+								'children' in item &&
```

Comment:
What are your thoughts about converting some of these type checks into our own type guard functions? It might be a little more readable since we can name the function something descriptive, and they'll be reusable.

Reply:
Some of these will get simpler (`typeof item === 'object'` will eventually be implied by the type) but I think type guards are a good idea here!